halo/hooks/object.cpp

```diff
@@ -146,116 +147,119 @@ extern "C" { // Demangle naming for ASM.
 
-    memcpy(&new_eqip_beh, game_defs[static_cast<int>(ObjectType::EQUIPMENT)], s);
+    memcpy(&new_eqip_beh, game_defs[etoi(ObjectType::EQUIPMENT)], S);
     new_eqip_beh.parent_definitions[0] = &new_obje_beh;
     new_eqip_beh.parent_definitions[1] = &new_item_beh;
     new_eqip_beh.parent_definitions[2] = &new_eqip_beh;
 
-    memcpy(&new_garb_beh, game_defs[static_cast<int>(ObjectType::GARBAGE)], s);
+    memcpy(&new_garb_beh, game_defs[etoi(ObjectType::GARBAGE)], S);
     new_garb_beh.parent_definitions[0] = &new_obje_beh;
     new_garb_beh.parent_definitions[1] = &new_item_beh;
     new_garb_beh.parent_definitions[2] = &new_garb_beh;
 
-    memcpy(&new_proj_beh, game_defs[static_cast<int>(ObjectType::PROJECTILE)], s);
+    memcpy(&new_proj_beh, game_defs[etoi(ObjectType::PROJECTILE)], S);
     new_proj_beh.parent_definitions[0] = &new_obje_beh;
     new_proj_beh.parent_definitions[1] = &new_proj_beh;
 
-    memcpy(&new_scen_beh, game_defs[static_cast<int>(ObjectType::SCENERY)], s);
+    memcpy(&new_scen_beh, game_defs[etoi(ObjectType::SCENERY)], S);
     new_scen_beh.parent_definitions[0] = &new_obje_beh;
     new_scen_beh.parent_definitions[1] = &new_scen_beh;
 
-    memcpy(&new_mach_beh, game_defs[static_cast<int>(ObjectType::MACHINE)], s);
+    memcpy(&new_mach_beh, game_defs[etoi(ObjectType::MACHINE)], S);
     new_mach_beh.parent_definitions[0] = &new_obje_beh;
     new_mach_beh.parent_definitions[1] = &new_devi_beh;
     new_mach_beh.parent_definitions[2] = &new_mach_beh;
 
-    memcpy(&new_ctrl_beh, game_defs[static_cast<int>(ObjectType::CONTROL)], s);
+    memcpy(&new_ctrl_beh, game_defs[etoi(ObjectType::CONTROL)], S);
     new_ctrl_beh.parent_definitions[0] = &new_obje_beh;
     new_ctrl_beh.parent_definitions[1] = &new_devi_beh;
     new_ctrl_beh.parent_definitions[2] = &new_ctrl_beh;
 
-    memcpy(&new_lifi_beh, game_defs[static_cast<int>(ObjectType::LIGHT_FIXTURE)], s);
+    memcpy(&new_lifi_beh, game_defs[etoi(ObjectType::LIGHT_FIXTURE)], S);
     new_lifi_beh.parent_definitions[0] = &new_obje_beh;
     new_lifi_beh.parent_definitions[1] = &new_devi_beh;
     new_lifi_beh.parent_definitions[2] = &new_lifi_beh;
 
-    memcpy(&new_plac_beh, game_defs[static_cast<int>(ObjectType::PLACEHOLDER)], s);
+    memcpy(&new_plac_beh, game_defs[etoi(ObjectType::PLACEHOLDER)], S);
     new_plac_beh.parent_definitions[0] = &new_obje_beh;
     new_plac_beh.parent_definitions[1] = &new_plac_beh;
 
-    memcpy(&new_ssce_beh, game_defs[static_cast<int>(ObjectType::SOUND_SCENERY)], s);
+    memcpy(&new_ssce_beh, game_defs[etoi(ObjectType::SOUND_SCENERY)], S);
     new_ssce_beh.parent_definitions[0] = &new_obje_beh;
     new_ssce_beh.parent_definitions[1] = &new_ssce_beh;
 
-    // Backup old def pointers for when the DLL needs to unload.
+    // Backup old array of def pointers for when the DLL needs to unload.
 
-    memcpy(&backup, game_defs, sizeof(ObjectBehaviorDefinition));
+    const size_t COPY_SIZE = OBJECT_DEF_PTR_ARRAY_ABS_SIZE;
```

Comment:
Why redefine this here when the constant at the top of the file exists? I'd argue `OBJECT_DEF_PTR_ARRAY_ABS_SIZE` makes the intent more clear here than copy size, especially considering the other names involved in the below `memcpy` call.